lib/cli/src/commands/run/wasi.rs

```diff
@@ -116,6 +146,47 @@ pub struct Wasi {
     pub deny_multiple_wasi_versions: bool,
 }
 
+/// Various triggers that will cause the runtime to take snapshot
+/// of the WASM state and store it in the snapshot file.
+#[derive(Debug, Clone, Copy, PartialEq, Eq, PartialOrd, Ord, Hash)]
+pub enum SnapshotTrigger {
```

Comment:
I'd add another option: `NonDeterministicCall`. Thoughts?

Reply:
Added this as a placeholder and understand what you are doing with this, it makes sense.
https://github.com/wasmerio/wasmer/pull/4263/commits/8f11ac78b00d3f07b774e776d20e54221dc9e33f